pkg/controller/vitessshard/reconcile_tablets.go

```diff
@@ -306,7 +306,7 @@ func vttabletSpecs(vts *planetscalev2.VitessShard, parentLabels map[string]strin
 				drain.SupportedAnnotation: "ensure that the tablet is not a master",
 			}
 			update.Annotations(&annotations, pool.Annotations)
-
+			update.Annotations(&annotations, backupLocation.Annotations)
```

Comment:
I think we need to watch out for the case when `backupLocation` is `nil`. Unlike vtbackup, we still reconcile vttablets when backups are not configured.